### DISCUSSION TITLE
SQLAlchemy Data model visualization

### DISCUSSION THREAD

**Discussion by Dicklesworthstone** [sqlalchemy/sqlalchemy#10588] [Ideas]
Github repo: https://github.com/Dicklesworthstone/sqlalchemy_data_model_visualizer

I made this for fun, but I think it could be useful for people. If your application has pretty complicated SQLalchemy data models (and the app I'm working on now has gotten super complicated, with dozens of models with complex interconnections), then this very easily generates an attractive SVG that you can open in the browser showing all your tables and how they are connected. You can be up and running this in less than a minute-- just [open it ](https://colab.research.google.com/drive/1np5kPvDtdhq138eLHOGINYuTUMJo_wrj?usp=sharing) in Google Colab, replace the sample data models with your own, and run the notebook, and you should see the SVG displayed almost immediately at the bottom.

I attempted to also generate an HTML file containing the embedded SVG as well as some JS to enable nice interactive hover-on effects (I wanted to highlight the border of the table you hover over and then change all the outgoing connections to other tables to be red, and all incoming connections to that table to be green), but it wasn't working and started turning into a slog after an hour. If anyone knows how to do this, please submit a PR!

I think it would be a tough sell to include this in the library itself because it does have graphviz as a dependency, which is pretty heavy.

**Comment by zzzeek** [comment]
we have a "Graphing" category on the wiki at https://github.com/sqlalchemy/sqlalchemy/wiki/UsageRecipes which at this point has some really old stuff there, we can gladly link to your project there

**Comment by Dicklesworthstone** [comment]
That would be great, thank you!
